service/src/main/java/bio/terra/workspace/service/admin/flights/cloudcontexts/gcp/SetupWorkingMapForUpdatedWorkspacesStep.java

```diff
@@ -0,0 +1,27 @@
+package bio.terra.workspace.service.admin.flights.cloudcontexts.gcp;
+
+import static bio.terra.workspace.service.workspace.flight.WorkspaceFlightMapKeys.UPDATED_WORKSPACES;
+
+import bio.terra.stairway.FlightContext;
+import bio.terra.stairway.Step;
+import bio.terra.stairway.StepResult;
+import bio.terra.stairway.exception.RetryException;
+import java.util.HashSet;
+
+/**
+ * Add an empty hashset to the working map that is to put all the updated workspace id in. This is
+ * for activity logging to record all the updated workspaces.
+ */
+public class SetupWorkingMapForUpdatedWorkspacesStep implements Step {
```

Comment:
Why do you need a step for this? Instead, can activity log skip writing activity log if UPDATED_WORKSPACES isn't in working map?

Reply:
so I can get it in the next step and not need to check whether the set exists everytime.